kernel/kernel-api/src/main/java/io/delta/kernel/types/ArrayType.java

```diff
@@ -35,6 +36,11 @@ public ArrayType(DataType elementType, boolean containsNull) {
         this.containsNull = containsNull;
     }
 
+    @Override
+    public byte getPromotionGroup(DataType dataType) {
```

Comment:
This is redundant code 
```
    @Override
    public byte getPromotionGroup(DataType dataType) {
        return typePromotionGroup;
    }
```
But need to be here because polymorphism does not play nicely with static variable, the alternative is to make `typePromotionGroup` an instance variable instead, neither option looks clean, perhaps code reviewers can point me a better alternative.